src/protocols/bdx/AsyncTransferFacilitator.h

```diff
@@ -0,0 +1,130 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/BitFlags.h>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeHolder.h>
+#include <protocols/bdx/BdxTransferSession.h>
+#include <transport/raw/MessageHeader.h>
+
+#pragma once
+
+namespace chip {
+namespace bdx {
+
+/**
+ * An abstract class with methods for handling BDX messages received from an ExchangeContext. Once a message is receievd, this
+ * class passes the message to the TransferSession to process the received message and gets the next output events from the
+ * TransferSession state machine and either sends a message accross the exchange or calls the HandleTransferSessionOutput virtual
+ * method to notify the subclass of the event generated. It keeps getting the next output event until it receieves an output event
+ * of type TransferSession::OutputEventType::kNone. For messages that are sent to the HandleTransferSessionOutput method, the
+ * subclass must call the NotifyEventHandled to notify the AsyncTransferFacilitator that the event has been handled and returns an
+ * error code for error cases or success.
```

Comment:
This isn't very clear to me.  If the subclass returns error, does it still need to call NotifyEventHandled?  What does "returns error" mean?  HandleTransferSessionOutput returns void.

In general, it seems like all the documentation related to NotifyEventHandled should live on the interface that defines that method, which is not this one?

Reply:
I will try to clarify a bit more and move the details to the interface

Reply:
done.